src/main/java/mate.academy/lib/Injector.java

```diff
@@ -1,13 +1,84 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
+        Field[] declaredFields = clazz.getDeclaredFields();
+        if (declaredFields.length == 0
+                || Arrays.stream(declaredFields)
+                .filter(f -> f.getType().isPrimitive())
+                .count() == declaredFields.length) {
+            return createNewInstance(clazz);
+        }
+        for (Field field : declaredFields) {
+            if (!field.getType().isPrimitive()
+                    && !findImplementation(field.getType()).isAnnotationPresent(Component.class)
+            ) {
```

Comment:
you are finding the implementation here for even initialized fields with no `@Inject` annotation? this is not correct, let's remove this as well

Reply:
just in case. if there is an annotation, but field was primitive. as i wrote above, primitive fields were not initialized without additional code